crates/rooch-types/src/crypto.rs

```diff
@@ -975,11 +976,14 @@ mod tests {
     fn test_ethereum_public_key_to_address() {
         let private_key = Secp256k1RecoverablePrivateKey::from_bytes(&[1u8; 32]).unwrap(); // use 1u8.
         let keypair: Secp256k1RecoverableKeyPair = private_key.into();
-        let address: RoochAddress = keypair.public().into();
-        assert_eq!(
-            address.to_string(),
-            "0x8c891976da9498ec1d3ff778a5d6c40c217d63cc8c48539c959f8b683eedf5a4"
-        );
+        let public_key = keypair.public();
+        let uncompressed = public_key.pubkey.serialize_uncompressed();
+        let uncompressed_64 = uncompressed[1..65].to_vec();
+        let hashed = keccak256(uncompressed_64);
+        let address_bytes = hashed[12..32].to_vec();
+        let address_str = format!("0x{}", hex::encode(address_bytes)); // Include "0x" prefix
+        let expected_address = "0x1a642f0e3c3af545e7acbd38b07251b3990914f1";
```

Comment:
This address has collision with an address already exists on Mainnet Ethereum.

Reply:
This code only implements key to address conversion. It means it's accepting a public key only. If someone wants to recover a public key from signature and message hash, they can do so and pass the recovered public key to the function.

The address here can be tested at https://toolkit.abdk.consulting/ethereum#key-to-address. Just input the public key `0x03a4f75d35449bb6c72757f3e2e765eecb6b15407d9f55678616b7ecf7e7e48211 ` and you get the same address `0x6A84EAE48a19f1ec014C627d4A1ffc50ce42c516 `.

The code works.